casanovo/data/datasets.py

```diff
@@ -84,7 +84,31 @@ def __getitem__(self, idx) -> Tuple[torch.Tensor, float, int, str]:
         spectrum = self._process_peaks(
             mz_array, int_array, precursor_mz, precursor_charge
         )
-        return spectrum, precursor_mz, precursor_charge, str(idx)
+        return (
+            spectrum,
+            precursor_mz,
+            precursor_charge,
+            self.get_spectrum_id(idx)[1],
+        )
+
+    def get_spectrum_id(self, idx: int) -> Tuple[str, str]:
+        """
+        Return the identifier of the MS/MS spectrum with the given index.
+
+        Parameters
+        ----------
+        idx : int
+            The index of the MS/MS spectrum within the SpectrumIndex.
+
+        Returns
+        -------
+        ms_data_file : str
+            The peak file from which the MS/MS spectrum was originally parsed.
+        identifier : str
+            The MS/MS spectrum identifier, per PSI recommendations.
+        """
+        with self.index:
```

Comment:
**question:** Could it be suboptimal that we always need to open a context manager to get a spectrum identifier? I don't really have a better solution though. What do you think @wfondrie?